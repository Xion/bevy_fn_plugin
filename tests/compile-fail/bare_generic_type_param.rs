#[macro_use] extern crate bevy_fn_plugin;

use bevy::prelude::*;

#[bevy_plugin]
fn generic_plugin<T>(_: &mut App) {}

fn main() {}
