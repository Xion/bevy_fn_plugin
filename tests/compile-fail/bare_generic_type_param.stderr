error: Invalid type parameters bounds

         = note: Generic type arguments to a #[bevy_plugin] function must be 'static
         = help: Add an explicit `T: 'static` bound, or another bound that implies it

 --> tests/compile-fail/bare_generic_type_param.rs:6:19
  |
6 | fn generic_plugin<T>(_: &mut App) {}
  |                   ^

warning: unused import: `bevy::prelude::*`
 --> tests/compile-fail/bare_generic_type_param.rs:3:5
  |
3 | use bevy::prelude::*;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
